src/js/events/events.js

```diff
@@ -1,98 +1,383 @@
+
+/**
+ * Fired when a new playlist item has been loaded into the player.
+*/
+export const PLAYLIST_ITEM = 'playlistItem';
+
+/**
+ * Fired when an entirely new playlist has been loaded into the player.
+*/
+export const PLAYLIST_LOADED = 'playlist';
+
+/**
+ * Fired when the list of available audio tracks is updated. Happens shortly after a playlist item starts playing.
+*/
+export const AUDIO_TRACKS = 'audioTracks';
+
+/**
+ * Fired when the active audio track is changed.
+    Happens in response to e.g. a user clicking the audio tracks menu or a script calling setCurrentAudioTrack().
+*/
+export const AUDIO_TRACK_CHANGED = 'audioTrackChanged';
+
+/**
+ * Fired when the playback rate has been changed.
+*/
+export const PLAYBACK_RATE_CHANGED = 'playbackRateChanged';
+
+// View Component Actions
+
+/**
+ * Fired when a click has been detected on the logo element.
+*/
+export const LOGO_CLICK = 'logoClick';
+
+// Model - Captions
+
+/**
+ * Fired when the list of available captions tracks changes.
+    This event is the ideal time to set default captions with the API.
+*/
+export const CAPTIONS_LIST = 'captionsList';
+
+/**
+ * Triggered whenever the active captions track is changed manually or via API.
+*/
+export const CAPTIONS_CHANGED = 'captionsChanged';
+
+// Provider Communication
+
+/**
+ * Fired the provider being utilized by JW Player for a particular media file is replaced by a new provider.
+*/
+export const PROVIDER_CHANGED = 'providerChanged';
+
+/**
+ * Triggered when a provider begins playback to signal availability of first frame.
+*/
+export const PROVIDER_FIRST_FRAME = 'providerFirstFrame';
+
+// UI Events
+
+/**
+ * Fired when user activity is detected on the targeted element.
+*/
+export const USER_ACTION = 'userAction';
+
+/**
+ * Fired when the instream adapter detects a click.
+*/
+export const INSTREAM_CLICK = 'instreamClick';
+
+/**
+ * Triggered when the player is resized to a width in a different breakpoint category.
+*/
+export const BREAKPOINT = 'breakpoint';
+
+// States
+
+/**
+ * The user pressed play, but sufficient data to start playback has not yet loaded.
+   The buffering icon is visible in the display.
+*/
+export const STATE_BUFFERING = 'buffering';
+
+/**
+ * Either playback has not started or playback was stopped due to a stop() call or an error.
+   In this state, either the play or the error icon is visible in the display.
+*/
+export const STATE_IDLE = 'idle';
+
+/**
+ * Playback has ended. The replay icon is visible in the display.
+*/
+export const STATE_COMPLETE = 'complete';
+
+/**
+ * The video is currently paused. The play icon is visible in the display.
+*/
+export const STATE_PAUSED = 'paused';
+
+/**
+ * The video is currently playing. No icon is visible in the display.
+*/
+export const STATE_PLAYING = 'playing';
+
+/**
+ * Playback was stopped due to an error.
+   In this state the error icon and a message are visible in the display.
+*/
+export const STATE_ERROR = 'error';
```

Comment:
The goal is to only have one string literal for each value, and use the constants through the projects (jwplayer and jwplayer-commercial). 

Having state and event constants makes sense. You use the state constants when setting state and event ones when listening to or triggering events. To avoid duplication define the state constants first and then set the events to those really every state should have its own event otherwise just keep the states and remove the duplicate events.

export const ERROR_STATE = 'error';
export const ERROR_EVENT = ERROR_STATE;